### ISSUE TITLE
Question: how can conda not find Python 3.5?

### ISSUE THREAD

**Issue by ijstokes** [conda/conda#2598]
I just don't understand what is going on here, especially since Python 3.5 is already installed and available in the environment.

```
((impyla_test)) root@ip-172-31-55-246:~# conda -V
conda 4.1.0rc3

((impyla_test)) root@ip-172-31-55-246:~# conda install jupyter notebook
Using Anaconda Cloud api site https://api.anaconda.org
Fetching package metadata .......
Solving package specifications ....
Error: Package missing in current linux-64 channels: 
  - conda-forge::python 3.5*

You can search for this package on anaconda.org with

    anaconda search -t conda conda-forge::python 3.5*

You may need to install the anaconda-client command line client with

    conda install anaconda-client
```

For completeness:

```
((impyla_test)) root@ip-172-31-55-246:~# conda info -a
Using Anaconda Cloud api site https://api.anaconda.org
Current conda install:

             platform : linux-64
        conda version : 4.1.0rc3
  conda-build version : 1.20.0
       python version : 2.7.11.final.0
     requests version : 2.9.1
     root environment : /opt/cloudera/parcels/Anaconda  (writable)
  default environment : /opt/cloudera/parcels/Anaconda/envs/impyla_test
     envs directories : /opt/cloudera/parcels/Anaconda/envs
        package cache : /opt/cloudera/parcels/Anaconda/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False

# conda environments:
#
impyla_test           *  /opt/cloudera/parcels/Anaconda/envs/impyla_test
root                     /opt/cloudera/parcels/Anaconda

sys.version: 2.7.11 |Anaconda 4.0.0 (64-bit)| (defaul...
sys.prefix: /opt/cloudera/parcels/Anaconda
sys.executable: /opt/cloudera/parcels/Anaconda/bin/python
conda location: /opt/cloudera/parcels/Anaconda/lib/python2.7/site-packages/conda
conda-build: /opt/cloudera/parcels/Anaconda/bin/conda-build
conda-convert: /opt/cloudera/parcels/Anaconda/bin/conda-convert
conda-develop: /opt/cloudera/parcels/Anaconda/bin/conda-develop
conda-env: /opt/cloudera/parcels/Anaconda/bin/conda-env
conda-index: /opt/cloudera/parcels/Anaconda/bin/conda-index
conda-inspect: /opt/cloudera/parcels/Anaconda/bin/conda-inspect
conda-metapackage: /opt/cloudera/parcels/Anaconda/bin/conda-metapackage
conda-pipbuild: /opt/cloudera/parcels/Anaconda/bin/conda-pipbuild
conda-server: /opt/cloudera/parcels/Anaconda/bin/conda-server
conda-sign: /opt/cloudera/parcels/Anaconda/bin/conda-sign
conda-skeleton: /opt/cloudera/parcels/Anaconda/bin/conda-skeleton
user site dirs: 

CIO_TEST: <not set>
CONDA_DEFAULT_ENV: /opt/cloudera/parcels/Anaconda/envs/impyla_test
CONDA_ENVS_PATH: <not set>
LD_LIBRARY_PATH: <not set>
PATH: /opt/cloudera/parcels/Anaconda/envs/impyla_test/bin:/usr/lib/jvm/java-7-openjdk-amd64/jre/bin::/usr/lib/jvm/java-7-openjdk-amd64/jre/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PYTHONHOME: <not set>
PYTHONPATH: <not set>

WARNING: could not import _license.show_info
# try:
# $ conda install -n root _license
```

and also:

```
((impyla_test)) root@ip-172-31-55-246:~# conda list
# packages in environment at /opt/cloudera/parcels/Anaconda/envs/impyla_test:
#
bitarray                  0.8.1                    py35_0  
sasl                      0.2.1                    py35_0    blaze
thrift_sasl               0.2.0                    py35_0    blaze
impyla                    0.13.8                   py35_0    conda-forge
ncurses                   5.9                           7    conda-forge
python                    3.5.1                         0    conda-forge
thriftpy                  0.3.7                    py35_0    conda-forge
tk                        8.5.19                        0    conda-forge
zlib                      1.2.8                         3    conda-forge
openssl                   1.0.2h                        1  
pip                       8.1.1                    py35_1  
ply                       3.8                      py35_0  
readline                  6.2                           2  
setuptools                21.2.1                   py35_0  
six                       1.10.0                   py35_0  
sqlite                    3.9.2                         0  
wheel                     0.29.0                   py35_0  
xz                        5.0.5                         1  
```

FWIW, the environment was created with:

```
conda create -c blaze -c conda-forge -n impyla_test python=3 thrift_sasl impyla
```

**Comment by mcg1969** [comment]
Is there a particular reason you're using 4.1.0rc3? Can you try dropping back to 4.0.7 and see if that works? We're still ironing out some bugs in the channel collision handling for 4.1.x. But I'll try and make sure that this is checked before we push out 4.1.

**Comment by mcg1969** [comment]
By the way, now that conda-forge is overlapping with defaults on fundamental packages like Python itself you may want to use `-c conda-forge` with care (regardless of the conda version). 

Please don't get me wrong, this bug shouldn't have happened, I just wanted to point out you did grab conda-forge's Python instead of ours.

**Comment by mcg1969** [comment]
The following works correctly with current `master`:
```
conda create -c blaze -c conda-forge -n impyla_test python=3 thrift_sasl impyla
conda install jupyter notebook
```

**Comment by mcg1969** [comment]
Oops, I forgot the `-n impyla_test` in the second command there. It works, but with a weird output (excerpt):
```
The following packages will be UPDATED:

    python:            3.5.1-0 conda-forge --> 3.5.1-0       conda-forge
```
I will investigate this but again it did work.

**Comment by mcg1969** [comment]
It looks like it's actually replacing the conda-forge python with defaults python. It's not clear to me why it's doing this, and of course it's giving the wrong channel for the new package, too.

**Comment by mcg1969** [comment]
This is actually fixed in the latest versions of conda, and is not related to #2791

**Comment by nnashok** [comment]
@mcg1969 , can you point the version this has been fixed in? I'm using 4.1.11 and still seeing this issue.

**Comment by mcg1969** [comment]
I recommend filing a new issue if you're still having problems.
